### DISCUSSION TITLE
OnCrosshairMove TimeScale issue

### DISCUSSION THREAD

**Discussion by christose** [tradingview/lightweight-charts#748] [General] [answered]
I have an issue with intervals with even values e.g. 2H, 4H. So when an even interval is selected, e.g. 2H, the object in OnCrosshairMove callback contains a time that is irrelevant with the data inside the chart, the data are for every 2 hours but the time in that object has the hour as an odd value so instead of the actual candle in the collection that has time e.g. 22:00:00, the time in the object has the time 21:00:00 that doesn't exist in the data collection or anywhere else.

**Comment by timocov** [comment, marked as answer]
Do you use the same Date's object's methods to see that time? Do you see the same timezone there? Can you provide an example (jsfiddle-like)?

**Reply by christose** [reply]
Mhm it's the timezone! When I insert the data in series are in UTC, the local time zone is utc+3 thus the data are presented with this timezone on the chart, however, the object's time in OnCrosshairMove callback is in UTC. Now I get it...

Any ideas on how we can do this properly? :)
Do I need to convert the utc time to the local time ?

**Reply by timocov** [reply]
It depends on what you're trying to achieve and solve.

> the data are presented with this timezone on the chart

I don't think that it's so since we don't use a local time anywhere, we're always using methods like `getUTC*` to retrieve some parts from the date. Probably you override some time formatters or so?

**Reply by christose** [reply]
Once again only by discussing it with you, you helped! :)
I've changed the logic and from the collection's object I use the unix timestamp instead of the datetime object which was messing with the timezone on conversion, so that solved the issue and It works fine now.

Something like this:
```
public OnCrosshairMove(obj: any): void {
    var time = obj.time;
    var priceObj = this.Data.find(x => x.Time == time); // Time is in the unix timestamp format
}
```

I need to keep in mind next time that one of javascript's big weaknesses is the date object :)

**Reply by timocov** [reply]
> that one of javascript's big weaknesses is the date object :)

Yeah, but not only JS I'm afraid 😂
